woocommerce/google-listings-and-ads
# API Pull - Expose Attribute mapping Rules Values

### Changes proposed in this Pull Request:



Part of https://github.com/woocommerce/google-listings-and-ads/issues/2146

As we move to a new approach for fetching products using the WC REST API, we need to make product's attribute mapping values accessible. Hence, this PR introduces a new property to the `wc/v3/products` endpoint, named `gla_attributes`. This property will contain the attribute values after applying all GLA Mapping Attribute rules, with GLA attributes taking precedence over mapping rules.

Also, I realized that most of the logic for mapping attribute rules resides within `WCProductAdapter`. In my view, moving this logic to the `AttributeManager` class would be more logical, given its existing handling of GLA attributes. Therefore, I've moved and adjusted most of the logic from the `WCProductAdapter` to the `AttributeManager` class. This adjustment will enable us to independently retrieve attributes from a product, without needing to create an instance of `WCProductAdapter`.

I was thinking of refactoring `WCProductAdapter` to use the new methods in the `AttributeManager` class, however, as we are not going to sync the attributes anymore I believe it's unnecessary to refactor that section as we will delete that part of the code in the next stages.

### Screenshots:




### Detailed test instructions:


1. Go to GLA -> Attribute -> Create some rules.
2. Make the following request `GET wc/v3/products?gla_syncable=1` or `GET wc/v3/products/YOUR_PRODUCT_ID?gla_syncable=1`
3. See that the response contains a new property named: `gla_attributes` with the correct values.

![image](https://github.com/woocommerce/google-listings-and-ads/assets/2488994/ac985dcb-4feb-4250-938b-14e8c37d3f6b)


4. Update the products by adding a GLA Attribute. This attribute should take precedence over any mapping rule applied to the same attribute.

![image](https://github.com/woocommerce/google-listings-and-ads/assets/2488994/d6819f86-c696-4601-ad1b-3a6a7843dace)

5. Remove the query parameter `gla_syncable=1` and the response should not contain `gla_attributes`.


### Additional details:


### Changelog entry

>
